### ISSUE TITLE
AttributeError: 'CloudManager' object has no attribute

### ISSUE THREAD

**Issue by venkat-pinisetti** [red-hat-storage/ocs-ci#3016]
Seems CloudManager clients are not yet implemented for s390x platform. Hence getting below error during test execution.

**Test case:** tests/manage/mcg/test_multi_region.py::TestMultiRegion::test_multiregion_bucket_creation

**Environment:**

OCP : 4.5
OCS : 4.6
Platform : s390x

**Log:**
```
request = <SubRequest 'cloud_uls_factory' for <Function test_multiregion_bucket_creation>>, cld_mgr = <ocs_ci.ocs.resources.cloud_manager.CloudManager object at 0x3ff825adb38>

    @pytest.fixture(scope='class')
    def cloud_uls_factory(request, cld_mgr):
        """
        Create an Underlying Storage factory.
        Calling this fixture creates a new underlying storage(s).
    
        Returns:
           func: Factory method - each call to this function creates
               an Underlying Storage factory
    
        """
>       return cloud_uls_factory_implementation(request, cld_mgr)

tests/conftest.py:1877: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

request = <SubRequest 'cloud_uls_factory' for <Function test_multiregion_bucket_creation>>, cld_mgr = <ocs_ci.ocs.resources.cloud_manager.CloudManager object at 0x3ff825adb38>

    def cloud_uls_factory(request, cld_mgr):
        """
        Create an Underlying Storage factory.
        Calling this fixture creates a new underlying storage(s).
    
        Args:
            request (object): Pytest built-in fixture
            cld_mgr (CloudManager): Cloud Manager object containing all connections to clouds
    
        Returns:
            func: Factory method - each call to this function creates
                an Underlying Storage factory
    
        """
        all_created_uls = {
            'aws': set(),
            'google': set(),
            'azure': set(),
            'ibmcos': set()
        }
    
        ulsMap = {
            'aws': cld_mgr.aws_client,
>           'google': cld_mgr.google_client,
            'azure': cld_mgr.azure_client,
            # TODO: Implement - 'ibmcos': cld_mgr.ibmcos_client
        }
E       AttributeError: 'CloudManager' object has no attribute 'google_client'

ocs_ci/ocs/resources/cloud_uls.py:31: AttributeError
```
**Auth File:**
```
[root@ocsvm2 ~]# cat ocs-ci/data/auth.yaml 
quay:
  access_token: <token>
AUTH:
  AWS:
    AWS_ACCESS_KEY_ID: <ID>
    AWS_SECRET_ACCESS_KEY: <SECRET>
```

**Comment by RazTamir** [comment]
@MeridianExplorer could you please take a look at this issue?
This is blocking IBM Z and I wonder what would it take to implement the missing part on their side?

@ebenahar if you have any input on this, please share

**Comment by MeridianExplorer** [comment]
@RazTamir, @venkat-pinisetti -
This problem happened because of an incomplete `auth.yaml` file.
CloudManager is dynamically creating its attributes based on the skeleton of the `auth.yaml` file, even if it is empty.
Thus, the file needs to follow this format, even if the credentials remain empty - 
```
AUTH:
  quay:
    access_token: 

  ipmi:
  - hostname: hostname
    console: console
    username: username
    password: password

  AWS:
    AWS_ACCESS_KEY_ID: 
    AWS_SECRET_ACCESS_KEY:
  GOOGLE:
    GOOGLE_APPLICATION_CREDENTIALS:
  AZURE:
    TENANT:
    CLIENT_ID:
    SECRET:
    SUBSCRIPTION_ID:
  IBMCOS:
    S3COMP: True
    AWS_ACCESS_KEY_ID:
    AWS_SECRET_ACCESS_KEY:
    ENDPOINT:
```

**Comment by venkat-pinisetti** [comment]
@MeridianExplorer, with empty values, testcase was skipped and also I can see warning message `Failed to fetch auth.yaml from ocs-ci-data`.

```
(.venv) [root@ocsvm2 ocs-ci]# cat data/auth.yaml 
AUTH:
  quay:
    access_token:

  ipmi:
  - hostname: hostname
    console: console
    username: username
    password: password

  AWS:
    AWS_ACCESS_KEY_ID:  
    AWS_SECRET_ACCESS_KEY: 
  GOOGLE:
    GOOGLE_APPLICATION_CREDENTIALS:
  AZURE:
    TENANT:
    CLIENT_ID:
    SECRET:
    SUBSCRIPTION_ID:
  IBMCOS:
    S3COMP: True
    AWS_ACCESS_KEY_ID:
    AWS_SECRET_ACCESS_KEY:
    ENDPOINT:
(.venv) [root@ocsvm2 ocs-ci]# run-ci tests/manage/mcg/test_multi_region.py::TestMultiRegion::test_multiregion_bucket_creation
Failed to fetch auth.yaml from ocs-ci-data
========================================================= test session starts =========================================================
platform linux -- Python 3.6.8, pytest-5.3.5, py-1.9.0, pluggy-0.13.1
rootdir: /root/ocs-ci, inifile: pytest.ini
plugins: reportportal-1.10.0, logger-0.5.1, metadata-1.10.0, html-2.1.1, marker-bugzilla-0.9.4, ordering-0.6
collected 1 item                                                                                                                      

tests/manage/mcg/test_multi_region.py::TestMultiRegion::test_multiregion_bucket_creation SKIPPED                                [100%]

========================================================== warnings summary ===========================================================
/root/.venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891
/root/.venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891
  /root/.venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: pytest_reportportal
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================================== 1 skipped, 2 warnings in 0.02s ====================================================
(.venv) [root@ocsvm2 ocs-ci]#
```
Since the testcase got skipped updated values as below. However failed with below error.
```
  AWS:
    AWS_ACCESS_KEY_ID: <ID>
    AWS_SECRET_ACCESS_KEY: <SECRET>
```
**Error:**
```
botocore.exceptions.ClientError: An error occurred (InvalidAccessKeyId) when calling the ListBuckets operation: The AWS Access Key Id you provided does not exist in our records.
```
Please suggest me next steps.

**Comment by bernerhat** [comment]
Hi @venkat-pinisetti ,
The skip occurs since you cant leave the auth file with no values, insert the credentials you do have to the skeleton provided by @MeridianExplorer :

``` AUTH:
  quay:
    access_token: 

  ipmi:
  - hostname: hostname
    console: console
    username: username
    password: password

  AWS:
    AWS_ACCESS_KEY_ID: <your id>
    AWS_SECRET_ACCESS_KEY: <your secret>
  GOOGLE:
    GOOGLE_APPLICATION_CREDENTIALS:
  AZURE:
    TENANT:
    CLIENT_ID:
    SECRET:
    SUBSCRIPTION_ID:
  IBMCOS:
    S3COMP: True
    AWS_ACCESS_KEY_ID:
    AWS_SECRET_ACCESS_KEY:
    ENDPOINT:
```

**Comment by venkat-pinisetti** [comment]
Hi @bernerhat , Unfortunately I dont have AWS credentials. Is there any other option.

**Comment by MeridianExplorer** [comment]
@venkat-pinisetti - There is no other option. The test requires access to AWS, and the usage of AWS credentials.
Without such credentials, it cannot run.
@ebenahar - Could you please weigh in?

**Comment by petr-balogh** [comment]
One more issue little bit related is this one:
https://github.com/red-hat-storage/ocs-ci/issues/3033

**Comment by venkat-pinisetti** [comment]
@MeridianExplorer Can you please elaborate a bit what type of AWS account required to execute these test cases and it would be good if you can share how are you guys handling in your Jenkins pipeline.

**Comment by petr-balogh** [comment]
IIRC @venkat-pinisetti, current logic depends on file data/auth.yaml
https://github.com/red-hat-storage/ocs-ci/blob/96271fe1b3024703f0729167c538a2eb0359e449/ocs_ci/utility/utils.py#L1694

So in our pipeline we have this sharedLib:
https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/-/blob/master/vars/createAuthConfig.groovy

Which we call before calling run-ci, so it will make sure that the file from jenkins credentials is taken copied to that location.

So all our auth.yaml file is stored in jenkins credentials and copied to needed place.

@MeridianExplorer or maybe I am wrong and it's not used like this in cloud manager, so please correct me if I am not correct as I remember there was also some other functionality to load this directly from S3.

**Comment by MeridianExplorer** [comment]
@venkat-pinisetti - The tests require an AWS account with permissions to create, delete, write and read to AWS S3 buckets. AFAIK, our team relies on a business account. Data storage on S3 is essential for some of the tests, thus requiring a payment method, too.

@petr-balogh - The current flow is very convoluted and confusing, since it was implemented in a sub-optimal manner because of time constraints.
https://github.com/red-hat-storage/ocs-ci/blob/60a986ef9ffc1d3cfd22cbd0484bf7410ee829ac/ocs_ci/ocs/resources/cloud_manager.py#L35-L55
The code first tries to update the cached auth data with `update_config_from_s3()`. This relies on OCS-CI's account information being found in the env variables, and then reaches our `ocs-ci-data` bucket on AWS, and pulls `auth.yaml` from there.
If it fails (i.e. - cases where the credentials aren't OCS-CI's), we try to load the credentials from `auth.yaml`.
Then, we rely on `auth.yaml` complying with the skeleton we have stored in our bucket, and described [in this comment abive](https://github.com/red-hat-storage/ocs-ci/issues/3016#issuecomment-701377322) to dynamically set the clients that the cloud manager offers.
